src/main/java/algorithms/curated170/medium/maze3/TheMazeIIIDijkstra.java

```diff
@@ -0,0 +1,124 @@
+package algorithms.curated170.medium.maze3;
+
+import java.util.Comparator;
+import java.util.PriorityQueue;
+
+public class TheMazeIIIDijkstra {
+
+    int[][] maze;
+    int[] hole;
+
+    public String findShortestWay(int[][] maze, int[] ball, int[] hole) {
+
+        this.maze = maze;
+        this.hole = hole;
+        return dijkstra(ball);
+    }
+
+    public String dijkstra(int[] ball) {
+
+        PriorityQueue<Point> pq = new PriorityQueue<>(new Comparator<Point>() {
+            @Override
+            public int compare(Point p1, Point p2) {
+                if (p1.distance == p2.distance) {
+                    return p1.path.compareTo(p2.path);
+                } else {
+                    return p1.distance - p2.distance;
+                }
+            }
+        });
+        int[][] directions = {{0, -1}, {1, 0}, {-1, 0}, {0, 1}};
+        char[] directionSymbols = {'l', 'd', 'u', 'r'};
+
+        pq.add(new Point(ball[0], ball[1], 0, ""));
+
+        while (!pq.isEmpty()) {
+
+            Point startPoint = pq.poll();
+
+            if (startPoint.x < 0) {
+                break;
+            }
```

Comment:
As in the previous question, I don't understand the function of this statement.

Reply:
> As in the previous question, I don't understand the function of this statement. Why do we check x but bot y?

So, does it work without it?

Reply:
Yeah, it was just a dirty residue of a code trial. It is my fault. Sorry about that.